typegen/src/utils.rs

```diff
@@ -31,18 +33,18 @@ pub fn ensure_unique_type_paths(types: &mut PortableRegistry) {
         // Compare existing groups to check which to add our type ID to.
         let mut added_to_existing_group = false;
         for group in groups_with_same_path.iter_mut() {
-            let ty_id_b = group[0]; // all types in group are same shape; just check any one of them.
-            let ty_b = types.resolve(ty_id_b).expect("ty exists");
-            if types_equal_extended_to_params(&ty.ty, ty_b) {
-                group.push(ty.id);
+            let other_ty_in_group_idx = group[0]; // all types in group are same shape; just check any one of them.
+            let other_ty_in_group = types.types.get(other_ty_in_group_idx).expect("ty exists");
```

Comment:
no need to not use `types.resolve()` here I guess? all it does is a `get` under the hood and then gives back `.ty` :)